2-mandatory/2-financial-times.js

```diff
@@ -5,77 +5,122 @@
     Implement the function below, which will return a new array containing only article titles which will fit.
 */
 function potentialHeadlines(allArticleTitles) {
-    // TODO
+  const charLimit = 65;
+  const articlesUnderLimit = [];
+
+  for (const articleTitle of allArticleTitles) {
+    if (articleTitle.length <= charLimit) {
+      articlesUnderLimit.push(articleTitle);
+    }
+  }
+  return articlesUnderLimit;
 }
 
 /*
     The editor of the FT likes short headlines with only a few words!
     Implement the function below, which returns the title with the fewest words.
-    (you can assume words will always be seperated by a space)
+    (you can assume words will always be separated by a space)
 */
 function titleWithFewestWords(allArticleTitles) {
-    // TODO
+  // TODO
+  let shortestTitle = "";
+  let shortestSpaceCount = 0;
+  for (const title of allArticleTitles) {
+    if (shortestTitle === "") {
+      shortestTitle = title;
+      shortestSpaceCount = title.split(" ").length - 1;
+    } else {
+      let currentSpaceCount = title.split(" ").length - 1;
+      if (currentSpaceCount < shortestSpaceCount) {
+        shortestTitle = title;
+        shortestSpaceCount = currentSpaceCount;
+      }
+    }
+  }
+  return shortestTitle;
 }
 
 /*
     The editor of the FT has realised that headlines which have numbers in them get more clicks!
     Implement the function below to return a new array containing all the headlines which contain a number.
     (Hint: remember that you can also loop through the characters of a string if you need to)
 */
+
+function containsNumbers(str) {
+  return /[0-9]/.test(str);
+}
+
 function headlinesWithNumbers(allArticleTitles) {
-    // TODO
+  let numberArticles = [];
+  for (const title of allArticleTitles) {
+    if (containsNumbers(title)) {
+      numberArticles.push(title);
+    }
+  }
+  return numberArticles;
 }
 
 /*
     The Financial Times wants to understand what the average number of characters in an article title is.
     Implement the function below to return this number - rounded to the nearest integer.
 */
 function averageNumberOfCharacters(allArticleTitles) {
-    // TODO
+  // TODO
+  let charCount = 0;
+  const articleTitleCount = allArticleTitles.length;
```

Comment:
Don't generally need to store `allArticlesTitles.length` in a variable. Absolutely fine to write `allArticleTitles.length` inline.